lib/dbservice_web/controllers/user_controller.ex

```diff
@@ -170,4 +176,45 @@ defmodule DbserviceWeb.UserController do
       |> render("show.json", user: user)
     end
   end
+
+  def get_user_sessions(conn, %{"user_id" => user_id, "quiz" => quiz_flag}) do
+    group_users = GroupUsers.get_group_user_by_user_id(user_id)
+
+    sessions = Enum.flat_map(group_users, &get_group_user(&1, quiz_flag))
+
+    render(conn, "user_sessions.json", session: sessions)
+  end
+
+  def get_user_sessions(conn, %{"user_id" => user_id}) do
+    get_user_sessions(conn, %{"user_id" => user_id, "quiz" => false})
+  end
+
+  defp get_group_user(group_user, quiz_flag) do
+    group_id = group_user.group_id
+
+    case Groups.get_group_by_group_id(group_id) do
+      nil -> []
+      group -> get_group(group, quiz_flag)
+    end
+  end
+
+  defp get_group(group, quiz_flag) do
+    child_id = group.child_id
+    batch = Batches.get_batch!(child_id)
```

Comment:
here we assume that type is always batch.. maybe its fine for now but later on we will have to expand to other types